untitled.js

```diff
@@ -327,529 +145,228 @@ function computeAttackML(ojn, charid, app, mod) {
+	        			} else {
+							let bmExtra=((bm !== 0) ? `+${bm}` :'');
+	        				out.impactstr += `${base}+${impactmod[0]}(d${sides}${bmExtra}): ${ir}`;
+	        			}
+			        }
+			    } 
 			}
 		}
-	} else {
-		var aspect = parseInt(myGet(aojn.slice(0, -4) + atk[3], charid, 0));
-		var aspstr = atk[3];
 	}
-	if (atk[4] == "missile") {
-	    if (wepname in missile_range) {
-	        aspect = state.MainGameNS.missi[1];
-	    } else {
-	        aspect = Math.round(aspect * parseFloat(state.MainGameNS.missi[1]))
-	    }
-	}
-
-	if (toke.get('bar3_value')) {
+	for (i = 0; i < base; i++) {
+		var ir = randomInteger(sides) + bm;
+		out.total += ir
+		if (out.impactstr.length > 2) {
+			out.impactstr += " + " + ir;
 
-		var pp = (parseInt(toke.get('bar3_value')) + parseInt(myGet(
-				'ENCUMBRANCE', defcharid, 0))) * 5;
+		} else {
+			let bmExtra=((bm !== 0) ? `+${bm}` :'');
+			out.impactstr += `${base}(d${sides}${bmExtra}): ${ir}`;
 
-	} else {
-		var pp = (parseInt(myGet('UNIVERSAL_PENALTY', defcharid, 0)) + parseInt(myGet(
-				'ENCUMBRANCE', defcharid, 0))) * 5;
+		}
 	}
+	
 
-	if (def[1] == "dodge") {
-	    var defml = 0;
-	    if ((atk[4] == "missile") && (wepname.indexOf("Bow") !== -1)) {
-		    defml = Math.round(parseInt(myGet("DODGE_ML", defcharid, 0))/2) + parseInt(def[2])- (pp);
-	    } else {
-	        defml = parseInt(myGet("DODGE_ML", defcharid, 0)) + parseInt(def[2])- (pp);
-	    };
-
-		var defstr = "";// "+" +  parseInt(myGet("DODGE_ML", defcharid, 0)) + "+" + parseInt(def[2])+ "-" + (pp);
-		var defwepname = "Dodge";
-
-	}
+	out.total += wepimpact.impact;
+	out.impactstr += " + "  + "<br>Weapon Impact: " + wepimpact.impact;
+	out.aspect = wepimpact.aspect;
+	return out;
 
-	if ((def[1] == "block") || (def[1] == "counterstrike")) {
-		var defwepname = msg.content.slice((msg.content.indexOf("WeaponName:") + 11));
-
-		if (defwepname.length > 3) {
-
-			var defwep = filterObjs(function(obj) {
-				obn = obj.get('name');
-				if (obn) {
-					if ((obn.indexOf("WEAPON_NAME")) !== -1
-							&& (obj.get("_characterid") == defcharid)
-							&& (obj.get("current") == defwepname)) {
-						return true;
-					} else {
-						return false;
-					}
-				} else {
-					return false;
-				}
-			});
+}
 
-			var ojn = defwep[0].get('name');
+function getWeaponImpact(repeating_weapon_name, charid, aspect = "H", missi = null) {
+	var out = {};
 
-			if (def[1] == "counterstrike") {
-			    var defml = parseInt(myGet(ojn.slice(0, -4) + "ML", defcharid, 0)) - pp + parseInt(myGet(ojn.slice(0, -4) + "ATK", defcharid, 0)) + parseInt(myGet(ojn.slice(0, -4) + "HM", defcharid, 0)) + parseInt(def[2]);
 
-			} else {
-			    var defml = parseInt(myGet(ojn.slice(0, -4) + "ML", defcharid, 0)) - pp + parseInt(myGet(ojn.slice(0, -4) + "DEF", defcharid, 0)) + parseInt(myGet(ojn.slice(0, -4) + "HM", defcharid, 0)) + parseInt(def[2]);
+	if (aspect == "H") {
 
+		var baspect = myGet(`${repeating_weapon_name}B`, charid, 0)
```

Comment:
Lines 234-257 could be tightened up a fair bit.
parse the three (well it should be four - Fire/Frost is excluded) aspects into a map from aspect abbreviation to number. Treat '-' as NaN (not a number) or NegInf (negative infinity).
Then add in a fifth element - 'H' -> max(P,E,B,F). Here's the sloppy bit - if statements to record which aspect is max. :/
last, return an object with the value from the map and the aspect.